cpp/src/decisiontree/batched-levelalgo/split.cuh

```diff
@@ -148,7 +148,7 @@ void initSplit(Split<DataT, IdxT>* splits, IdxT len, cudaStream_t s) {
 template <typename DataT, typename IdxT, int TPB = 256>
 void printSplits(Split<DataT, IdxT>* splits, IdxT len, cudaStream_t s) {
   auto op = [] __device__(Split<DataT, IdxT> * ptr, IdxT idx) {
-    printf("quesval = %f, colid = %d, best_metric_val = %f, nLeft = %d\n",
+    printf("quesval = %e, colid = %d, best_metric_val = %e, nLeft = %d\n",
```

Comment:
minor nitpick: stray printf?

Reply:
Do you mean to suggest use of `std::cout` instead? We can not use logger here since, when `printSplits` is called the printing should be guaranteed irrespective of the current logging level.

Reply:
Is `printSplits` used by a public API, or is it a debugging tool?

Reply:
Its for debugging only.

Reply:
Just realized another compulsion for using `printf`. We can not use `std::cout` or logger.

Reply:
ah... I missed the fact that this is a device method! Sorry about that. I don't have any issues with keeping this method around for debugging.